data/data/openstack/topology/private-network.tf

```diff
@@ -128,15 +129,15 @@ resource "openstack_networking_floatingip_associate_v2" "api_fip" {
 }
 
 resource "openstack_networking_router_v2" "openshift-external-router" {
-  count               = local.create_router
+  count               = var.external_network != "" ? local.create_router : 0
   name                = "${var.cluster_id}-external-router"
   admin_state_up      = true
-  external_network_id = data.openstack_networking_network_v2.external_network.id
+  external_network_id = join("",data.openstack_networking_network_v2.external_network.*.id)
```

Comment:
should this be "external_network_id = join("",data.openstack_networking_network_v2.external_network.[0].id)"
does it matter?

Reply:
I am not sure what "*" does, but by adding a count, it makes `data.openstack_networking_network_v2.external_network` a list. You should be able to dereference the id with just `data.openstack_networking_network_v2.external_network[0].id` I think

Reply:
I like the join trick though,that is clever. You can also use a dynamic block here if you want to be fancy

Reply:
I think the "*" allows for list of things with more than one, for example 
join("",data.openstack_networking_network_v2.external_network.[0].id) will only work with the first element of the list, if it had 2 or more, [1] would not be added.  but since we are only using a count of 0 or 1 external networks, I think we can just use [0] instead of "*"

Reply:
> I like the join trick though,that is clever. You can also use a dynamic block here if you want to be fancy

can you give example.  Most of this join stuff is needed because if data.openstack_networking_network_v2.external_network[0] is not created then terraform barfs here because basically data.openstack_networking_network_v2.external_network.[0].id is undefined. and terraform can't handle that.

Reply:
Hmm, actually now that I am looking at it, you cant because its not a code block. In which case, this is the best solution